rayon-core/src/thread_pool/mod.rs

```diff
@@ -252,6 +252,12 @@ impl ThreadPool {
         // We assert that `self.registry` has not terminated.
         unsafe { spawn::spawn_fifo_in(op, &self.registry) }
     }
+
+    /// Returns the current threadpool.
+    pub fn current() -> ThreadPool {
+        let registry = crate::registry::Registry::current();
+        ThreadPool { registry }
```

Comment:
We need to at least call `registry.increment_terminate_count()` here, otherwise `Drop` will signal threads to exit.

Reply:
Sure, thank you for spotting it!